app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdUpdate.kt

```diff
@@ -1,17 +1,29 @@
 package org.simple.clinic.scanid
 
 import com.spotify.mobius.Next
-import com.spotify.mobius.Next.noChange
 import com.spotify.mobius.Update
 import org.simple.clinic.mobius.dispatch
+import org.simple.clinic.mobius.next
+import org.simple.clinic.scanid.ShortCodeValidationResult.Failure
+import org.simple.clinic.scanid.ShortCodeValidationResult.Success
 
 class ScanSimpleIdUpdate : Update<ScanSimpleIdModel, ScanSimpleIdEvent, ScanSimpleIdEffect> {
   override fun update(model: ScanSimpleIdModel, event: ScanSimpleIdEvent): Next<ScanSimpleIdModel, ScanSimpleIdEffect> {
     return when (event) {
       ShowKeyboard -> dispatch(HideQrCodeScannerView)
       HideKeyboard -> dispatch(ShowQrCodeScannerView)
       ShortCodeChanged -> dispatch(HideShortCodeValidationError)
-      is ShortCodeValidated -> noChange()
+      is ShortCodeValidated -> shortCodeValidated(model, event)
+      is ShortCodeSearched -> next(model.shortCodeChanged(event.shortCode), ValidateShortCode(event.shortCode))
     }
   }
+
+  private fun shortCodeValidated(model: ScanSimpleIdModel, event: ShortCodeValidated): Next<ScanSimpleIdModel, ScanSimpleIdEffect> {
+    val effect = when (event.result) {
+      Success -> OpenPatientShortCodeSearch(model.shortCode?.shortCodeText.orEmpty())
```

Comment:
Since if it's a `Success`, we know for certain that the `shortCode` property in the model will not be `null`, we could use the `!!` and skip the null handling.

Reply:
Done